lib/stream.js

```diff
@@ -115,18 +119,18 @@ export class Stream<T> {
     static combine<T>(streams: Array<Stream<T>>): Stream<Array<T>> {
         return new Stream((update, finish) => {
             let combined = new Array(streams.length);
-            let updated = new Set();
-            let finished = new Set();
+            let updated = new ImmutableSet();
+            let finished = new ImmutableSet();
             streams.forEach((s, i) => {
                 s.values.attach((v) => {
                     combined[i] = v;
-                    updated.add(i);
+                    updated = updated.add(i);
```

Comment:
I started looking at this more closely... using ImmutableSet here is actually pretty useless and we don't really win, the sets are small

I am reversing the changes here